### ISSUE TITLE
Build failing with error MSB4062 (gets fixed with cache clean)

### ISSUE THREAD

**Issue by JuditRose** [dotnet/winforms#417]
* .NET Core Version: 3.0.100-preview-010024
* Have you experienced this same bug with .NET Framework?: No

Repro:
Run build from command line (or terminal in VS code)
(with clean working directory, having successful builds before, same master head version building successfully on other machines)

**Actual behavior:**  Build output:
E:\repos\winforms\.dotnet\sdk\3.0.100-preview-010024\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.DefaultItems.targets(127,5): error MSB4062: The "CheckIfPackageReferenceShouldBeFrameworkReference" task could not be loaded from the assembly E:\repos\winforms\.dotnet\sdk\3.0.100-preview-010024\
Sdks\Microsoft.NET.Sdk\targets\..\tools\net46/Microsoft.NET.Build.Tasks.dll.  Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask. [E:\repos\winforms
\src\System.Design\src\System.Design.Facade.csproj]
E:\repos\winforms\.dotnet\sdk\3.0.100-preview-010024\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.DefaultItems.targets(127,5): error MSB4062: The "CheckIfPackageReferenceShouldBeFrameworkReference" task could not be loaded from the assembly E:\repos\winforms\.dotnet\sdk\3.0.100-preview-010024\
Sdks\Microsoft.NET.Sdk\targets\..\tools\net46/Microsoft.NET.Build.Tasks.dll.  Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask. [E:\repos\winforms
\pkg\Microsoft.Private.Winforms.csproj]

Build FAILED.

E:\repos\winforms\.dotnet\sdk\3.0.100-preview-010024\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.DefaultItems.targets(127,5): error MSB4062: The "CheckIfPackageReferenceShouldBeFrameworkReference" task could not be loaded from the assembly E:\repos\winforms\.dotnet\sdk\3.0.100-preview-010024\
Sdks\Microsoft.NET.Sdk\targets\..\tools\net46/Microsoft.NET.Build.Tasks.dll.  Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask. [E:\repos\winforms
\src\System.Design\src\System.Design.Facade.csproj]
E:\repos\winforms\.dotnet\sdk\3.0.100-preview-010024\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.DefaultItems.targets(127,5): error MSB4062: The "CheckIfPackageReferenceShouldBeFrameworkReference" task could not be loaded from the assembly E:\repos\winforms\.dotnet\sdk\3.0.100-preview-010024\
Sdks\Microsoft.NET.Sdk\targets\..\tools\net46/Microsoft.NET.Build.Tasks.dll.  Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask. [E:\repos\winforms
\pkg\Microsoft.Private.Winforms.csproj]
    0 Warning(s)
    2 Error(s)

![image](https://user-images.githubusercontent.com/16646604/52447795-b3450380-2ae6-11e9-9cf0-000ad0a846f8.png)

**Expected behavior:** Build is successful.

It got fixed after these caches cleaned:
 - winforms\ .dotnet folder removed
 - winforms\artifacts folder removed
 - c:\Users\myuser\ .nuget folder removed

**Comment by merriemcgaw** [comment]
@JuditRose thank for the investigation and steps to fix this situation should it come up again.

**Comment by danmosemsft** [comment]
I have no idea about this specific issue, but note that buildtools was never incremental with respect to the build tools, and I assume this did not change in the move to arcade tools. In other words, if your build pulls down updated tools, there is always a chance you need to do `git clean -fdx` and build again. I hope you would never need to clean the global cache though.

**Comment by Logerfo** [comment]
This is happening to me. I'm using .NET Core 3.1. `git clean -fdx` doesn't solve it.

**Comment by RussKie** [comment]
@Logerfo we need more information in order to triage it.

**Comment by VosWouter87** [comment]
Installing the .NET 5 SDK preview fixed this for me.

**Comment by Youssef1313** [comment]
I have seen this in a branch new Hello, world console app.

https://github.com/dotnet/sdk/issues/9632 suggests to run `dotnet build-server shutdown`.

It didn't work for me, but may work for others.

For my case, this was caused by `MSBuildSdksPath` environment variable being set to a version different than what I specified in a `global.json`. I had to update the path to match the SDK version in `global.json`
